tests/unit_tests/CMakeLists.txt

```diff
@@ -106,65 +106,57 @@ foreach(domain ${TARGET_DOMAINS})
   endif()
 
   if(ENABLE_MKLCPU_BACKEND)
-    add_dependencies(test_main_${domain}_ct onemkl_${domain}_mklcpu)
     list(APPEND ONEMKL_LIBRARIES_${domain} onemkl_${domain}_mklcpu)
   endif()
 
   if(ENABLE_MKLGPU_BACKEND)
-    add_dependencies(test_main_${domain}_ct onemkl_${domain}_mklgpu)
     list(APPEND ONEMKL_LIBRARIES_${domain} onemkl_${domain}_mklgpu)
   endif()
 
   if(domain STREQUAL "blas" AND ENABLE_CUBLAS_BACKEND)
-    add_dependencies(test_main_${domain}_ct onemkl_${domain}_cublas)
     list(APPEND ONEMKL_LIBRARIES_${domain} onemkl_${domain}_cublas)
   endif()
 
   if(domain STREQUAL "blas" AND ENABLE_ROCBLAS_BACKEND)
-    add_dependencies(test_main_${domain}_ct onemkl_${domain}_rocblas)
     list(APPEND ONEMKL_LIBRARIES_${domain} onemkl_${domain}_rocblas)
   endif()
 
   if(domain STREQUAL "blas" AND ENABLE_NETLIB_BACKEND)
-    add_dependencies(test_main_${domain}_ct onemkl_${domain}_netlib)
     list(APPEND ONEMKL_LIBRARIES_${domain} onemkl_${domain}_netlib)
   endif()
 
   if(domain STREQUAL "blas" AND ENABLE_PORTBLAS_BACKEND)
-    add_dependencies(test_main_${domain}_ct onemkl_${domain}_portblas)
     list(APPEND ONEMKL_LIBRARIES_${domain} onemkl_${domain}_portblas)
   endif()
 
   if(domain STREQUAL "lapack" AND ENABLE_CUSOLVER_BACKEND)
-    add_dependencies(test_main_${domain}_ct onemkl_${domain}_cusolver)
     list(APPEND ONEMKL_LIBRARIES_${domain} onemkl_${domain}_cusolver)
   endif()
 
   if(domain STREQUAL "lapack" AND ENABLE_ROCSOLVER_BACKEND)
-    add_dependencies(test_main_${domain}_ct onemkl_${domain}_rocsolver)
     list(APPEND ONEMKL_LIBRARIES_${domain} onemkl_${domain}_rocsolver)
   endif()
   
   if(domain STREQUAL "rng" AND ENABLE_CURAND_BACKEND)
-    add_dependencies(test_main_${domain}_ct onemkl_${domain}_curand)
     list(APPEND ONEMKL_LIBRARIES_${domain} onemkl_${domain}_curand)
   endif()
 
   if(domain STREQUAL "rng" AND ENABLE_ROCRAND_BACKEND)
-    add_dependencies(test_main_${domain}_ct onemkl_${domain}_rocrand)
     list(APPEND ONEMKL_LIBRARIES_${domain} onemkl_${domain}_rocrand)
   endif()
 
   if(domain STREQUAL "dft" AND ENABLE_CUFFT_BACKEND)
-    add_dependencies(test_main_${domain}_ct onemkl_${domain}_cufft)
     list(APPEND ONEMKL_LIBRARIES_${domain} onemkl_${domain}_cufft)
   endif()
 
   if(domain STREQUAL "dft" AND ENABLE_ROCFFT_BACKEND)
-    add_dependencies(test_main_${domain}_ct onemkl_dft_rocfft)
```

Comment:
I'd rather that we don't remove these dependencies as part of this PR and follow the same format for portfft. If we really want to improve this it should be discussed in a different PR IMO.

Reply:
ok, reverted here https://github.com/FMarno/oneMKL/pull/1/commits/d2fb2bc96d8b5fe66d4e34b744ede1ecda953710

Reply:
How about we also add the dependency for portFFT backend to keep it consistent for now?